test/e2e/node/container_readiness_during_termination.go

```diff
@@ -0,0 +1,259 @@
+			pod, err := podClient.Get(ctx, pod.Name, metav1.GetOptions{})
+			if err != nil {
+				// If pod is not found, it means it was fully deleted
+				// This is expected during termination, so we can stop monitoring
+				if apierrors.IsNotFound(err) {
+					framework.Logf("Pod was deleted during monitoring - this is expected during termination")
+					return true, nil
+				}
+				return false, err
+			}
+
+			// Check if pod is still terminating
+			if pod.DeletionTimestamp == nil {
+				return false, fmt.Errorf("pod should be terminating")
+			}
+
+			// Count ready containers
+			readyContainers := 0
+			for _, status := range pod.Status.ContainerStatuses {
+				if status.Ready {
+					readyContainers++
+				}
+			}
+
+			framework.Logf("Pod status: %d/%d containers ready", readyContainers, len(pod.Status.ContainerStatuses))
+
+			// The key test: we should see 1/2 containers ready at some point during termination
+			if readyContainers == 1 {
+				readinessUpdated = true
+				framework.Logf("SUCCESS: Readiness updated to %d/2 during termination (expected 1/2)", readyContainers)
+				return true, nil
+			}
+
+			// If all containers are dead, we're done
+			if readyContainers == 0 {
+				return true, nil
+			}
+
+			return false, nil
+		})
+
+		framework.ExpectNoError(err, "failed to monitor pod termination")
+
+		ginkgo.By("Verifying readiness was updated during termination")
+		gomega.Expect(readinessUpdated).To(gomega.BeTrueBecause("container readiness should be updated when containers die during termination"))
+
+		// Additional verification: check that the pod eventually gets fully terminated
+		ginkgo.By("Waiting for pod to be fully terminated")
+		err = wait.PollUntilContextTimeout(ctx, 5*time.Second, 3*time.Minute, true, func(ctx context.Context) (bool, error) {
+			_, err := podClient.Get(ctx, pod.Name, metav1.GetOptions{})
+			if err != nil {
+				// Pod not found means it's fully terminated
+				return true, nil
+			}
+			return false, nil
+		})
+		framework.ExpectNoError(err, "pod should be fully terminated")
+	})
+
+	ginkgo.It("should update readiness when container crashes during termination", func(ctx context.Context) {
+		ginkgo.By("Creating a pod with container that will crash")
+		podName := "test-pod-liveness-readiness-" + string(uuid.NewUUID())
+		pod := &v1.Pod{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      podName,
+				Namespace: ns,
+			},
+			Spec: v1.PodSpec{
+				TerminationGracePeriodSeconds: func() *int64 { v := int64(120); return &v }(),
+				Containers: []v1.Container{
+					{
+						Name:  "test-container",
+						Image: imageutils.GetE2EImage(imageutils.Agnhost),
+						Args:  []string{"sh", "-c", "sleep 10; echo 'Container crashing...'; exit 1"},
+						Lifecycle: &v1.Lifecycle{
+							PreStop: &v1.LifecycleHandler{
+								Exec: &v1.ExecAction{
+									Command: []string{"sh", "-c", "sleep 30"},
```

Comment:
it will be WAY MORE reliable test to let it be something huge... like infinity for all practical puroses. Same as TerminationGracePeriodSeconds. 

Then your test will wait for one container becomeing NotReady and another keeps being Ready - terminating.

Then you can kill the Pod with force flag.

Reply:
The way test implemented - it will likely flake at some point